### DISCUSSION TITLE
How to handle v-field-template input?

### DISCUSSION THREAD

**Discussion by IlyaPetrovM** [directus/directus#7470] [Archive]
Hello! 

I want to create interface with `v-field-template` (It is connected with [that issue](https://github.com/directus/directus/discussions/7400#discussioncomment-1185006)). 

Unfortunately, I can't save any values using this interface. The field stays blank after update. And `handleChange` function doesn't react on field pasting. 

My code:

### src/index.js
```js
import InterfaceComponent from './interface.vue';

export default {
	id: 'formInterface',
	name: 'formInterface',
	description: 'This is my custom interface!',
	icon: 'box',
	component: InterfaceComponent,
	types: ['string'],
};
```

### src/interface.vue
```js
<template>
<v-field-template :collection="collection" 
                  @input="handleChange($event)" />
</template>

<script>
export default {
	emits: ['input'],
	props: {
		value: String,
                collection: String,
	},
	methods: {
		handleChange(event) {
			this.$emit('input', event);
                        console.log(event);
		},
	},
};
</script>
```

### Version of Directus:
9.0.0.rc-90

**Comment by rijkvanzanten** [comment]
> The field stays blank after update.

I believe this might be due to the fact that you're not passing the value back into the v-field-template component using `:value="value"`

**Reply by IlyaPetrovM** [reply]
No, nothing changed after that correction.

And handle doesn't react on pasting:
![image](https://user-images.githubusercontent.com/6277090/129797300-8646f03f-b596-4bd5-9662-2611bd5704b9.png)

**Reply by rijkvanzanten** [reply]
I'm confused, what are we looking at in your screenshot?

**Reply by IlyaPetrovM** [reply]
It is `v-form-template`. I've created Custom Interface using code above, and now I'm testing it with collection "authors" (Book, First Name etc)

**Reply by rijkvanzanten** [reply]
Oh my b, I was wondering what the red arrows are supposed to mean 🙂

**Reply by IlyaPetrovM** [reply]
"When I click to any Item it pastes to input above". And handle function doesn't react on pasting

**Reply by rijkvanzanten** [reply]
> "When I click to any Item it pastes to input above".

That's how the v-field-template component is supposed to work 🙂 It's basically an input to make creating display template strings easier to create, like `Title: {{ article.title }}`. When you click one of the fields in the dropdown, it simply appends the field contained in `{{ }}`

**Reply by IlyaPetrovM** [reply]
`@update:modelValue` event works for me!

I've changed code:
### src/interface.vue
```
<v-field-template :collection="collection" 
                  :value="value"
                  @update:modelValue="handleChange($event)" />
</template>
...
```
String with fields stores in collection,

BUT

when I open item, the string disappears.

**Reply by rijkvanzanten** [reply]
Ohhhhhhh right right, i mistook this for an interface. Interfaces use `:value/@input`, components use `:model-value/@update:modelValue`

**Reply by rijkvanzanten** [reply]
Try replacing that `:value` with `:model-value`

**Reply by IlyaPetrovM** [reply]
Thank you! It works!
But with `emits: ['input']` (why not `emits: ['update:modelValue']` ?...)

working code:
```js
<template>
<v-field-template :collection="collection" 
                  :model-value="value"
                  @update:modelValue="handleChange($event)" />
</template>

<script>
export default {
	emits: ['input'],
	props: {
		value: String,
                collection: String,
	},
	methods: {
		handleChange(e) {
			this.$emit('input', e);
                        console.log(e);
		},
	},
};
</script>
```

**Reply by rijkvanzanten** [reply]
Because _your_ custom interface has to emit an `input` event 🙂
